tests/testthat/test-methods.R

```diff
@@ -58,9 +58,35 @@ test_that("as.data.frame.lints", {
     line = c("", "a <- 1"),
     linter = c("", "custom_linter"),
     stringsAsFactors = FALSE)
-  
+
   expect_equal(
     df,
     exp
-  )  
+  )
+})
+
+test_that("summary.lints works", {
+  good <- "if (1 > 0){
+      print(\"good\")
+    }"
+  bad <- "if(1>0){
+      print  ('bad')
+    }
+  "
+  file <- tempfile()
+  on.exit(unlink(file))
+
+  writeLines(good, con = file, sep = "\n")
+  expect_silent(out <- summary(lint(file)))
```

Comment:
I think this test would be more clear if you did something like

```r
no_lints <- lint(
"if (1 > 0){
   print(\"good\")
}")
no_lint_summary <- summary(no_lints)
# ... tests

has_lints <- lint(
"if(1>0){
  print  ('bad')
}")

has_lint_summary <- summary(has_lints)
# ... tests
```

Also I would explicitly pass linters to `lint()` rather than relying on the default ones. The defaults can change over time so they could potentially break this test if new linters are added.

Reply:
Good call. Just cleaned this up.